### DISCUSSION TITLE
-region and -channel option ?

### DISCUSSION THREAD

**Discussion by liuhaitao827** [dlemstra/Magick.NET#776] [Help] [answered]
Hi, I am trying to implement following commands in Magick.Net

_magick redeye1.jpg -region "36x36+120+150" -channel r -fx "(hypot(abs(138-i),abs(168-j))<=18 && p.r/((p.g+p.b)/2)>1.5)?(p.g+p.b)/2:p.r" redeyecls.jpg_

But, I don't find 'region' or 'channel' methods in Magick.Net. Any advice? 
Thank you in advance!

**Comment by dlemstra** [comment, marked as answer]
The `-channel` is an extra `Channel` argument to the methods so you will need to do this `image.Fx((hypot(abs(138-i),abs(168-j))<=18 && p.r/((p.g+p.b)/2)>1.5)?(p.g+p.b)/2:p.r", Channels.Red)`

And with the following `readSettings` you can read only an area of the image:

```C#
var readSettings = new MagickReadSettings
{
    ExtractArea = new MagickGeometry(120, 150, 36, 36),
};
```

**Reply by liuhaitao827** [reply]
This is helpful, thank you!
